src/scripts/main.js

```diff
@@ -1 +1,65 @@
 'use strict';
+
+const doc = document.querySelector('body');
+
+const firstPromise = new Promise((resolve, reject) => {
+  doc.addEventListener('click', () => {
+    resolve();
+  });
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve, reject) => {
+  doc.addEventListener('click', () => {
+    resolve();
+  });
+
+  doc.addEventListener('contextmenu', () => {
+    resolve();
+  });
+});
+
+const thirdPromise = new Promise((resolve, reject) => {
+  let leftClick = false;
+  let rightClick = false;
+
+  doc.addEventListener('click', () => {
+    leftClick = true;
+
+    if (leftClick && rightClick) {
+      resolve();
+    }
+  });
+
+  doc.addEventListener('contextmenu', (e) => {
+    e.preventDefault();
```

Comment:
Good job on preventing the default action for the context menu event in `thirdPromise`. This avoids unwanted browser behavior.